### PR TITLE
Fix to have all deck DSL functions to respect DPI settings

### PR DESCRIPTION
**Pull request by felixleong**
## Bug Description

When you change the `Deck::dpi` parameter (say, from the default 300 to 100), all DSL functions draws the cards as if it was still at 300 dpi. Causing the card image to clip.

This is caused by `arg_loader.rb` not sending the `dpi` parameter to the convert_units call.

Related to this issue, the `text()` call would treat all fonts as if it was on 96dpi instead of whatever the `Deck`'s dpi parameter. Hence you'd have to bump up the text size by 3 (So what you'd expect as a 12pt font on a normal word processor, you'll have to specify in a Squib layout to be 36pt). The current fix would fixes that issue at the cost of backwards compatibility.

## Side Effects/Compatibility Issue

As mentioned above, any pre-existing Squib projects being coded prior to this fix would see the text blowing up by 300% (assuming the default DPI of 300).

## Additional Feature Added

The "pt" unit is added so that people can use it to specify the size of a `text_embed` image relative to the surrounding font sizes (instead of relying on native pixel values that doesn't scale w/ the user's DPI setting.

## Remaining TODO

- [x] ~~save_pdf currently isn't fully aware of the Deck's DPI setting. So if you set the DPI to 100, the cards would appeared shrinked in the print out~~

### CODE REVIEW

File: lib/squib/args/unit_conversion.rb [hunk 1]
```diff
@@ -13,6 +13,8 @@ def parse(arg, dpi=300)
         case arg.to_s.rstrip
         when /in$/ # ends with "in"
           arg.rstrip[0..-2].to_f * dpi
+        when /pt$/ # ends with "in"
+          arg.rstrip[0..-2].to_f * dpi / 72
```
Comment (andymeneely) [inline lib/squib/args/unit_conversion.rb]: Can you make the 72 a constant?

**Comment by andymeneely** [comment]
Totally didn't know about context.resolution - I always wondered why the font sizes were weird. Thanks!

If you can't get the PDF issues fixed soon, I'd be happy to merge this as is now and put the pdf issue on the backlog.

File: lib/squib/args/unit_conversion.rb (hunk 1 above)
Comment (felixleong) [inline lib/squib/args/unit_conversion.rb]: Done. Have plugged the constant into `save_pdf` as well as it explains the code better as well.

**Comment by felixleong** [comment]
Have found what's the problem with my PDF problems and it's not a bug: it's because I forgot to specify the `width` and `height` parameters for `save_pdf()` in my code.

The code should be fine, might want to update the samples if everything's OK.

**Comment by andymeneely** [comment]
I'm definitely am going to get this into the code base soon...

**Comment by andymeneely** [comment]
This finally got merged!! Thank you again.

**Pull request closed without merging**